File: examples/python/lifecycle/lifecycle.py
from nightona import ListSandboxesQuery, Nightona, SandboxListSortDirection, SandboxListSortField, SandboxState


def main():
    nightona = Nightona()

    print("Creating sandbox")
    sandbox = nightona.create()
    print("Sandbox created")

    _ = sandbox.set_labels(
        {
            "public": "true",
        }
    )

    print("Stopping sandbox")
    nightona.stop(sandbox)
    print("Sandbox stopped")

    print("Starting sandbox")
    nightona.start(sandbox)
    print("Sandbox started")

    print("Getting existing sandbox")
    existing_sandbox = nightona.get(sandbox.id)
    print("Get existing sandbox")

    response = existing_sandbox.process.exec('echo "Hello World from exec!"', cwd="/home/nightona", timeout=10)
    if response.exit_code != 0:
        print(f"Error: {response.exit_code} {response.result}")
    else:
        print(response.result)

    for sb in nightona.list(
        ListSandboxesQuery(
            limit=10,
            labels={"env": "dev"},
            states=[SandboxState.STARTED],
            sort=SandboxListSortField.CREATEDAT,
            order=SandboxListSortDirection.DESC,
        )
    ):
        print(sb.id)

    print("Removing sandbox")
    nightona.delete(sandbox)
    print("Sandbox removed")


if __name__ == "__main__":
    main()

File: examples/ruby/lifecycle/lifecycle.rb
# frozen_string_literal: true

require 'nightona'

nightona = Nightona::Nightona.new

puts 'Creating sandbox'
sandbox = nightona.create
puts 'Sandbox created'

puts 'Replacing sandbox labels'
sandbox.labels = { public: true }
puts "Sandbox labels: #{sandbox.labels}"

puts 'Stopping sandbox'
nightona.stop(sandbox)
puts "Sandbox #{sandbox.state}"

puts 'Starting sandbox'
nightona.start(sandbox)
puts "Sandbox #{sandbox.state}"

puts 'Getting existing sandbox'
sandbox = nightona.get(sandbox.id)
puts 'Retrieved existing sandbox'

response = sandbox.process.exec(command: 'echo "Hello World from exec!"', cwd: '/home/nightona', timeout: 10)
if response.exit_code == 0
  puts response.result
else
  puts "Error: #{response.exit_code} #{response.result}"
end

sandboxes = nightona.list.to_a
puts "Total sandboxes count: #{sandboxes.size}"

puts "Printing sandboxes[0] -> id: #{sandboxes.first.id} state: #{sandboxes.first.state}" unless sandboxes.empty?

puts 'Removing sandbox'
nightona.delete(sandbox)
puts 'Sandbox removed'
